File: 1100/1490C/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        long long a;
        cin >> a;
        bool flag = false;

        for (long long i = 1; i * i * i <= a; i++) 
        {
            long long rem = a - i * i * i;
            if (rem <= 0) 
                continue;

            // Compute cube root of remainder
            long long c = cbrt(rem);

            // Check if c^3 equals rem (to avoid floating precision issues)
            if (c * c * c == rem || (c + 1) * (c + 1) * (c + 1) == rem) 
            {
                flag = true;
                break;
            }
        }

        if (flag)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}

File: 1100/1490C/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            long a = sc.nextLong();
            boolean flag = false;
            
            for(long i=1;i*i*i<=a;i++)
            {
                if(a-i*i*i <= 0)
                    continue
                else
                {
                    if(Math.cbrt(a-i*i*i) == (long)(Math.cbrt(a-i*i*i)))
                    {
                        flag = true;
                        break;
                    }
                }
            }
            
            if(flag)
                System.out.println("YES");
            else
                System.out.println("NO");
        }
    }
}